core/src/main/java/cz/xtf/core/openshift/ClusterVersionInfo.java

```diff
@@ -20,22 +23,35 @@
 
 @Slf4j
 public class ClusterVersionInfo {
-    // version must be in format major.minor.micro (4.8.13) or major.minor (4.8)
+    // version must contain major.minor (4.8)
+
+    private static final String OCP3_CLIENTS_URL = "https://mirror.openshift.com/pub/openshift-v3/clients";
+    private static final String OCP4_CLIENTS_URL = "https://mirror.openshift.com/pub/openshift-v4";
+
     private static final Pattern versionPattern = Pattern
-            .compile("^(\\d+\\.\\d+)(\\.\\d+)?(-\\d+\\.nightly-\\d{4}-\\d{2}-\\d{2}-\\d{6})?$");
-    private final String openshiftVersion;
-    private final Matcher versionMatcher;
+            .compile("(^(\\d+\\.\\d+).*)");
+    private String openshiftVersion;
+    private String openshiftVersionURL;
+    private Matcher versionMatcher;
 
     ClusterVersionInfo() {
-        if (StringUtils.isNotEmpty(OpenShiftConfig.version())) {
-            // manually configured version in config
-            openshiftVersion = OpenShiftConfig.version();
-        } else {
-            // try to detect version from cluster
-            openshiftVersion = detectClusterVersionFromCluster();
+        //get version from config
+        openshiftVersion = OpenShiftConfig.version();
+        versionMatcher = (openshiftVersion != null) ? versionPattern.matcher(openshiftVersion) : null;
+        if (versionMatcher != null && versionMatcher.matches()) {
+            getClientUrlBasedOnOcpVersion();
+        }
+
+        if (getClientUrl() != null && getOpenshiftVersion() != null) {
+            return;
         }
 
-        versionMatcher = openshiftVersion != null ? validateConfiguredVersion(openshiftVersion) : null;
+        //fallback - get version from cluster
+        openshiftVersion = detectClusterVersionFromCluster();
+        versionMatcher = (openshiftVersion != null) ? versionPattern.matcher(openshiftVersion) : null;
+        if (versionMatcher != null && versionMatcher.matches()) {
```

Comment:
I think we don't need to validate version from cluster.

Reply:
we don't really need that but I'm using version matcher's result as part of URL building procedure (to get majorMinor)